src/gretel_trainer/trainer.py

```diff
@@ -47,7 +47,7 @@ def __init__(
         self.cache_file = self._get_cache_file(cache_file)
         self.model_type = model_type
 
-        if self.overwrite:
+        if self.overwrite and self.model_type is not None:
             logger.debug(json.dumps(self.model_type.config, indent=2))
```

Comment:
Would it be worth logging something about "running with auto-config" in the case where `self.model_type is None`, or fine as-is?

Reply:
Yes, something to the effect of "Automatically selecting optimal model config" would be great.